mkosi/__init__.py

```diff
@@ -1776,6 +1776,7 @@ def finalize_default_initrd(
         *(["--output-dir", str(output_dir)] if output_dir else []),
         *(["--workspace-dir", str(config.workspace_dir)] if config.workspace_dir else []),
         *(["--cache-dir", str(config.cache_dir)] if config.cache_dir else []),
+        *(["--cache-key", config.cache_key]),
```

Comment:
This passes the finalized value with the specifiers replaced to the CLI call instead of the original template, but I'm not sure how else to implement this.